gemini/tuning/gemini_supervised_finetuning_text_classification.ipynb

```diff
@@ -0,0 +1,2295 @@
+        "    \"max_output_tokens\": 10,\n",
+        "    \"epochs\": epochs,\n",
+        "    \"lrm\": learning_rate_multiplier, # learning rate multiplier\n",
+        "    \"adapter_size\": adapter_size\n",
+        "}\n",
+        "\n",
+        "log_experiment_run_vertexai(\n",
+        "    experiment_name=EXPERIMENT_NAME,\n",
+        "    run_name=\"<your-experiment-run-name\",\n",
+        "    metrics=metrics_tuned_gemini,\n",
+        "    project=PROJECT_ID,\n",
+        "    location=LOCATION,\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "6cIFYQ-3fcuE"
+      },
+      "source": [
+        "## 5.  Evaluation comparisons\n",
+        "\n",
+        "To assess the performance of your experiments in Vertex AI, you have two primary options. You can programmatically retrieve a comprehensive table (pandas DataFrame) containing all experiments and their associated metrics for in-depth analysis. Alternatively, Vertex AI offers a user-friendly visual UI enabling you to compare experiments, select specific runs for side-by-side comparisons, and gain rapid insights. For detailed instructions on both approaches, refer to the [Vertex AI documentation on evaluation comparisons](https://cloud.google.com/vertex-ai/docs/experiments/compare-analyze-runs)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "vF948vqJfcuE"
+      },
+      "outputs": [],
+      "source": [
+        "df_experiments = get_experiments_data_frame_sample(\n",
+        "    experiment=EXPERIMENT_NAME,\n",
+        "    project=PROJECT_ID,\n",
+        "    location=LOCATION\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "8dtTMBKnfcuE"
+      },
+      "outputs": [],
+      "source": [
+        "df_experiments"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bUNM5LJA1OZq"
+      },
+      "source": [
+        "\n",
+        "\n",
+        "> In our experiments with this dataset the most performant model was achieved by tuning Gemini 1.0 Pro @002, with the below parameters:\n",
+        "\n",
+        "```\n",
+        "epochs=6, learning_rate_multiplier= 1, and adapter_size=4\n",
+        "```"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "pZzNV7fnuJsy"
+      },
+      "source": [
+        "## [Optional] 6. Heuristics for Computing Confidence Scores\n",
+        "\n",
+        "Due to the multitask essence of LLMs computing confidence scores is not as straightforwad as it is with traditional predictive AI. Gemini models do not expose logprobs for the time being. However, the below snippets provide some options to use as a proxy for confidence scores in your predictions. You can expand these options to your own use cases and needs.\n",
```

Comment:
```suggestion
        "Due to the multitask essence of LLMs computing confidence scores is not as straightforward as it is with traditional predictive AI. Gemini models do not expose logging problems for the time being. However, the below snippets provide some options to use as a proxy for confidence scores in your predictions. You can expand these options to your own use cases and needs.\n",
```

Reply:
logprobs is not logging but are the probabilities of the tokens.

Reply:
Ok, can you spell this out in the text?